File: server/schemas/group.py
"""Group schemas per ADDENDUM §R.3."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class TeacherShort(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    full_name: str


class GroupCreate(BaseModel):
    name: str = Field(min_length=2, max_length=200)
    description: str | None = Field(default=None, max_length=2000)


class GroupUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=2, max_length=200)
    description: str | None = Field(default=None, max_length=2000)
    is_active: bool | None = None


class GroupOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    description: str | None = None
    teachers: list[TeacherShort] = []
    student_count: int = 0
    is_active: bool
    created_at: datetime


class GroupMemberAdd(BaseModel):
    user_id: int


class GroupTeacherAssign(BaseModel):
    teacher_id: int

File: client/src/types/group.ts
export interface TeacherShort {
  id: number;
  full_name: string;
}

export interface GroupCreate {
  name: string;
  description?: string | null;
}

export interface GroupUpdate {
  name?: string | null;
  description?: string | null;
  is_active?: boolean | null;
}

export interface GroupOut {
  id: number;
  name: string;
  description: string | null;
  teachers: TeacherShort[];
  student_count: number;
  is_active: boolean;
  created_at: string;
}

export interface GroupMemberAdd {
  user_id: number;
}

export interface GroupTeacherAssign {
  teacher_id: number;
}
